src/features/api-keys/lib.ts

```diff
@@ -102,6 +121,15 @@ export const createApiKey = async (
 	options: ApiKeyOptions = {},
 ): Promise<string> => {
 	options.apiKey ??= randomstring.generate();
+	const { name, description } = getKeyMetadata(req.body);
+
+	if (!options.name) {
+		options.name = name;
+	}
+	if (!options.description) {
+		options.description = description;
+	}
```

Comment:
Do we intend to block setting empty stings?
if not then we should
```suggestion
	if (options.name != null) {
		options.name = name;
	}
	if (options.description != null) {
		options.description = description;
	}
```

Reply:
Shouldn't we block `empty` strings?

Reply:
We've gone through this before -- `string?` is *very* problematic because it actually has three states: `null/undefined`, `''` and some non-zero length string but for most intents and purposes the first two states mean the same thing: "i have no value". So I think ensuring empty string do not make it to the database is sensible.

Reply:
Why should we?
Have we been doing this historically?

Reply:
Well I see we have been blocking empty stings already.... so nevermind...

Reply:
Arguably, the correct way to handle this is to introduce a check on the term/fact type that constrains at least the name to a length greater than 3-4 characters.